app/src/main/java/com/ncnf/event/Event.java

```diff
@@ -53,11 +66,40 @@ public Event(String name, Date date, Location location, EventType type, PubPriv
     public void setDate(Date date) { this.date = date; }
     public void setLocation(Location location) { this.location = location; }
     public void setAttendees(List<String> attendees) {
-        this.attendees = new ArrayList<>(attendees);
+        this.attendees = new ArrayList<String>();
+        for(int i = 0; i < attendees.size(); ++i) {
+            this.attendees.add(attendees.get(i));
+        }
         numOfAttendees = attendees.size();
     }
+    public void setUuid(UUID uuid) {this.uuid = uuid; }
     public void setDescription(String description)  { this.description = description; }
 
     public void setNewOrganizer(Organizer newOwner) { this.organizer = newOwner; }
 
+    public static Event toEvent(String f) {
+
+        String[] s = f.split("@");
+
+        if(s[0].equals("PUBLIC")) {
+            return PublicEvent.toEvent(s);
+        }
+        else if(s[0].equals("PRIVATE")) {
+            return PrivateEvent.toEvent(s);
+        }
+
+        return null;
+    }
```

Comment:
There exists a serializer in Java https://docs.oracle.com/javase/8/docs/api/java/io/Serializable.html